### DISCUSSION TITLE
Getting an attribute error while inserting list-of-lists data into a MongoDB database using Mongoz

### DISCUSSION THREAD

**Discussion by prashikdewtale10** [dymmond/mongoz#48] [Q&A] [answered]
I'm encountering an issue while trying to insert student scores in a nested array format [['202405', 85.45], ['202406', 90.14]] 
using the Mongoz ODM for MongoDB. 
The insertion operation fails, getting Attribute error as 'list' object has no attribute 'items' on line number 175 of mongoz\core\db\docuemnts\base.py
and I'm unsure if it's due to a bug in the library or a mistake in my implementation.
i am using latest version of mongoz.

class implementation
```
import mongoz
from mongoz.core.db import fields

class Student(mongoz.Document):
    student_scores: List[Union[str, float]] = fields.ArrayList()
```

I expect the student document to be inserted with the scores data in the specified format, 
but it seems Mongoz is having trouble serializing the nested lists.

Any help would be greatly appreciated!

**Comment by tarsil** [comment]
@prashikdewtale10 sorry for taking some time but maybe you are having something internally not declared properly? So, I literally used your example as a test here https://github.com/dymmond/mongoz/blob/main/tests/models/manager/test_array_insert.py and it is passing without any issues.

Another thing aside, I saw your profile and it looks great. I saw your readme and you are using [Esmerald](https://esmerald.dev) in your example and that makes us very very very happy. Thank you so much for the support.

**Comment by tarsil** [comment]
@prashikdewtale10 apologies, I was able to replicate! Nice catch, it was an internal issue when converting decimals checking for lists. The latest version `0.10.3` should fix that. Thank you for this.

**Reply by prashikdewtale10** [reply, marked as answer]
Thank you so much for the support and thank you for quick release.

**Comment by tarsil** [comment]
@prashikdewtale10 is it solved? Btw. What do you think of Esmerald?

**Reply by prashikdewtale10** [reply]
Yes it is solved.
We are exploring Esmerald and creating application to serve api's and following best practices mentioned by you in documentation.
our team is bit curious about how to do queries on embedded documents using mongoz for performing CRUD operations on embedded documents.

**Reply by tarsil** [reply]
Ok apparently this is a thing ahahah. Embedded documents in Mongoz are simple Pydantic BaseModel which means no CRUD operations can be done there but if this is something that apparently people want, then it will be added
